src/main/java/org/generationcp/middleware/service/DataImportServiceImpl.java

```diff
@@ -429,6 +430,28 @@ public void processExperimentalDesign(final Workbook workbook, final String prog
 
 	}
 
+	@Override
+	public void addEntryTypeVariableIfNotExists(final Workbook workbook, final List<MeasurementVariable> measurementVariables,
+		final String programUUID) {
+		final List<MeasurementVariable> combinedVariables = new ArrayList<>();
+		combinedVariables.addAll(workbook.getConditions());
```

Comment:
You're adding the created ENTRY_TYPE variable in the factors list so I'm assuming that if it already exists, it's stored in the factors list. Are there instances where the ENTRY_TYPE variable is found in the conditions list?